transaction-service/src/main/java/pl/adamboguszewski/transactionservice/application/Transaction.java

```diff
@@ -0,0 +1,58 @@
+package pl.adamboguszewski.transactionservice.application;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Transaction {
+
+    private final List<Product> products;
+
+    private final double totalPrice;
+
+    private final String transactionId;
+
+    private final String checkoutId;
+
+    private final LocalTime transactionTime;
+
+    private final LocalDate transactionDate;
+
+    public Transaction(List<Product> products, String transactionId,
+                       long totalPrice, String checkoutId,
+                       LocalTime transactionTime, LocalDate transactionDate) {
+        this.transactionId = transactionId;
+        this.products = new ArrayList<>();
+        this.products.addAll(products);
```

Comment:
If product has a list with objects like our transaction here this will blow up. Leave this for now and make // [TODO]: